backend.go

```diff
@@ -14,11 +14,11 @@ var (
 type Backend interface {
 	// Authenticate a user. Return smtp.ErrAuthUnsupported if you don't want to
 	// support this.
-	Login(username, password string) (User, error)
+	Login(username, password string, info ConnInfo) (User, error)
```

Comment:
Is it possible to make this the first argument instead? And also a pointer?

Reply:
Okay, not making it a pointer is probably fine, seeing how `tls.ConnectionState` is passed around. Maybe rename to `smtp.ConnectionState` for consistency?

Reply:
Sure.